lib/src/main/java/com/auth0/android/lock/internal/configuration/ApplicationFetcher.java

```diff
@@ -25,122 +25,131 @@
 package com.auth0.android.lock.internal.configuration;
 
 import android.net.Uri;
+import android.os.AsyncTask;
+
 import androidx.annotation.NonNull;
-import android.util.Log;
 
 import com.auth0.android.Auth0;
 import com.auth0.android.Auth0Exception;
 import com.auth0.android.authentication.AuthenticationException;
-import com.auth0.android.callback.AuthenticationCallback;
+import com.auth0.android.callback.Callback;
+import com.auth0.android.request.HttpMethod;
+import com.auth0.android.request.RequestOptions;
+import com.auth0.android.request.ServerResponse;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.reflect.TypeToken;
-import com.squareup.okhttp.Callback;
-import com.squareup.okhttp.OkHttpClient;
-import com.squareup.okhttp.Request;
-import com.squareup.okhttp.Response;
 
 import org.json.JSONException;
 import org.json.JSONObject;
 import org.json.JSONTokener;
 
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.lang.reflect.Type;
+import java.nio.charset.Charset;
 import java.util.List;
 
 public class ApplicationFetcher {
 
     private static final String JSONP_PREFIX = "Auth0.setClient(";
-    private static final String TAG = ApplicationFetcher.class.getSimpleName();
 
     private final Auth0 account;
-    private final OkHttpClient client;
 
     /**
      * Helper class to fetch the Application information from Auth0 Dashboard.
      *
      * @param account the Application details to build the request uri.
-     * @param client  the OKHttpClient instance to use for the CDN request.
      */
-    public ApplicationFetcher(@NonNull Auth0 account, @NonNull OkHttpClient client) {
+    public ApplicationFetcher(@NonNull Auth0 account) {
         this.account = account;
-        this.client = client;
     }
 
     /**
      * Fetch application information from Auth0
      *
      * @param callback to notify on success/error
      */
-    public void fetch(@NonNull AuthenticationCallback<List<Connection>> callback) {
-        makeApplicationRequest(callback);
+    public void fetch(@NonNull Callback<List<Connection>, Auth0Exception> callback) {
+        FetchTask task = new FetchTask(account);
+        task.execute(callback);
     }
 
-    private void makeApplicationRequest(final AuthenticationCallback<List<Connection>> callback) {
-        Uri uri = Uri.parse(account.getConfigurationUrl()).buildUpon().appendPath("client")
-                .appendPath(account.getClientId() + ".js").build();
+    private static class FetchTask extends AsyncTask<Callback<List<Connection>, Auth0Exception>, Void, List<Connection>> {
 
-        Request req = new Request.Builder()
-                .url(uri.toString())
-                .build();
+        private final Auth0 account;
 
-        client.newCall(req).enqueue(new Callback() {
```

Comment:
We will no longer use the OkHttp client directly, but through the interface exposed by the `Auth0` class. We still have to do the error handling, as we're doing here.

The logic in this class has been moved to a "background task" `AsyncTask` where the network request will be safely executed.